src/som/interpreter/nodes/MessageSendNode.java

```diff
@@ -626,4 +643,33 @@ public ExpressionNode asMateNode() {
       return new MateGenericMessageSendNode(this);
     }
   }
+
+  public static class CascadeMessageSendNode
+      extends ExpressionNode {
+
+    @Child private ExpressionNode receiver;
+    final @Children private ExpressionWithReceiverNode[] messages;
+
+    public CascadeMessageSendNode(final ExpressionNode receiver,
+        final ExpressionWithReceiverNode[] messages, final SourceSection source) {
+
+      super(source);
+      this.receiver = receiver;
+      this.messages = messages;
+    }
+
+    @Override
+    @ExplodeLoop
+    public Object executeGeneric(final VirtualFrame frame) {
+
+      int i;
+      Object receiver = this.receiver.executeGeneric(frame);
+
+      for (i = 0; i < this.messages.length - 1; i++) {
```

Comment:
hm, I missed that. Think that's rather hard to comprehend. Would be better to be explicit and say `messages.length - 1` instead of using `i`